pkg/virt-controller/watch/drain/disruptionbudget/disruptionbudget.go

```diff
@@ -406,10 +406,18 @@ func (c *DisruptionBudgetController) execute(key string) error {
 	}
 
 	for i := range pdbs {
+		if vmi.IsFinal() {
+			if deleteErr := c.deletePDB(key, pdbs[i], vmi); deleteErr != nil {
+				err = deleteErr
```

Comment:
I think it would be better to put the vmi.IsFinal check inside the controller sync loop, I think its strange that we continue to iterate to the next PDB even when we receive an error.